### PR TITLE
Adding "Guidance"

### PR DESCRIPTION
**Pull request by lxxxvi**
@nethad

As discussed, here's a proposal for a "Guidance" to the exercise.

What's new?

* Added a `README.md` that describes the content of the exercise.
* Added a `GUIDANCE.md` with in-depth instructions that can be followed.
* Added empty files `customer.rb` and `order.rb`, as it is "best practice" to have classes in separate files.
* Improved data in CSV files to have more meaningful data (= less confusion).

GUIDANCE.md is quite extensive, may be too extensive? I'm looking forward to hearing your opinion!

### CODE REVIEW

File: GUIDANCE.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+In order to work with the input data, we'd like to re-structure it into an Array. Why?
+
+Having an Array, we can treat each **line** in a **loop**, which makes working with the data easier.
+
+```ruby
+customer_lines = customers_file.split("\n")
+```
+
+We tell ruby to `.split` the string using the delimiter `\n`. After that we have an Array with String elements. Now go and print the 13th (or any other) element from that array:
```
Comment (nerdinand) [inline GUIDANCE.md]: Maybe re-iterate what `\n` is?

File: GUIDANCE.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+customer_id = customer_cells[0]
+first_name = customer_cells[1]
+last_name = customer_cells[2]
+location = customer_cells[3]
+```
+
+With this setup, we can start using the variables as usual. For example we could print customer info like this:
+
+```ruby
+puts "Customer #{id}: #{first_name} #{last_name} (#{location})"
```
Comment (nerdinand) [inline GUIDANCE.md]: Maybe add an example output of this line?
e.g. `Customer 13: Shae Gildersleeve (Mumbai)`

File: GUIDANCE.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+```
+
+## Excercise
```
Comment (nerdinand) [inline GUIDANCE.md]:
```suggestion
## Exercise
```

File: GUIDANCE.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+| `order.rb`          | This class has to be implemented entirely.  |
+
+The CSV files are
+
+| Filename       | Comment                                               |
+|----------------|-------------------------------------------------------|
+| `customer.csv` | Contains customer data.                               |
+| `order.csv`    | Contains order data, with reference to customer data. |
+
+### Running the script
```
Comment (nerdinand) [inline GUIDANCE.md]:
```suggestion
### Running the code
```

File: GUIDANCE.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+Please note that the following approach is just _one of many_. It is NOT the only way how to work with CSV-input data.
+
+#### Milestone 1: The `Customer` class
+
+The `Customer` class (or "model") will be defined in `customer.rb`.
+
+1. Create a **constructor** that takes all customer data (`id`, `first_name`, etc.) as individual input arguments.
+2. Assign each input argument to **instance variables** and define corresponding _getter methods_.
+3. Then define a `print_info` method, that prints a nicely formatted customer information, for example `"Customer 13: Shae Gildersleeve (Zurich)"`
```
Comment (nerdinand) [inline GUIDANCE.md]: Wasn't it Mumbai? ;)

File: GUIDANCE.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+4. Finally, use the `Customer` class in `OrderAnalysis`, and remove obsolete lines.
+
+After these steps, the result should be the **same** as before:
```
Comment (nerdinand) [inline GUIDANCE.md:120]: But the initial code in `order_analysis.rb` doesn't print the location? Maybe we should add that there?

File: README.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
+[Back to the original README](README.md)
```
Comment (nerdinand) [inline README.md]: Is this link needed?

**Review by nerdinand — CHANGES_REQUESTED** [review]
Great stuff! I think it's good to have so much guidance, especially if you're doing the exercise on your own without a coach...

File: GUIDANCE.md (hunk 5 above)
Comment (lxxxvi) [reply GUIDANCE.md]: 👍 🙈

File: GUIDANCE.md (hunk 6 above)
Comment (lxxxvi) [reply GUIDANCE.md:120]: Yes, very good find. I added it to the code.

**Comment by lxxxvi** [comment]
@nerdinand Thanks for the review. I added the suggestions and squashed my mess into one commit.

**Review by nerdinand — APPROVED** [review]

**Review by hharen — APPROVED** [review]
Cool, I really like the 'milestones' and 'what you'll learn' parts. Guidances helps a lot, awesome supermario ;)

**Pull request merged**